### PR TITLE
Return in case of tcp connection error

### PR DESCRIPTION
**Pull request by zehortigoza**

### CODE REVIEW

**Comment by lucasdemarchi** [comment]
@edersondisouza review and apply please

**Comment by lucasdemarchi** [comment]
@zehortigoza LGTM

File: main.cpp [hunk 1]
```diff
@@ -412,6 +408,13 @@ void Mainloop::handle_tcp_connection()
     add_fd(fd, tcp, EPOLLIN);
 
     log_debug("Received tcp connection on fd %d", fd);
+    return;
+
+calloc_error:
+    close(fd);
```
Comment (edersondisouza) [inline main.cpp]: There's a trick here: `close()` may change `errno`, so we could lose track of original error.

Comment (zehortigoza) [inline main.cpp]: done

**Pull request merged**